src/plugins/vis_builder/public/application/components/data_source_select.tsx

```diff
@@ -36,6 +39,11 @@ export const DataSourceSelect = () => {
       onChange={(option) => {
         const foundOption = indexPatterns.filter((s) => s.id === option.id)[0];
         if (foundOption !== undefined && typeof foundOption.id === 'string') {
+          getUsageCollector().reportUiStats(
```

Comment:
Not sure what the implementation underneath is and if this is already happening, but should this be in a requestIdleCallback after the dispatch to not block user feedback on selection?

(same for other metrics)